src/dbnode/storage/series/types.go

```diff
@@ -173,7 +173,36 @@ type QueryableBlockRetriever interface {
 	// Flushes may occur and change the actual block state while iterating
 	// through this snapshot, so any logic using this function should take this
 	// into account.
-	BlockStatesSnapshot() map[xtime.UnixNano]BlockState
+	BlockStatesSnapshot() ShardBlockStateSnapshot
+}
+
+// ShardBlockStateSnapshot represents a snapshot of a shard's block state at
+// a moment in time.
+type ShardBlockStateSnapshot struct {
+	bootstrapped bool
+	snapshot     BootstrappedBlockStateSnapshot
+}
+
+// NewShardBlockStateSnapshot constructs a new NewShardBlockStateSnapshot.
+func NewShardBlockStateSnapshot(
+	bootstrapped bool,
+	snapshot BootstrappedBlockStateSnapshot,
+) ShardBlockStateSnapshot {
+	return ShardBlockStateSnapshot{
+		bootstrapped: bootstrapped,
+		snapshot:     snapshot,
+	}
+}
+
+// Snapshot returns a BootstrappedBlockStateSnapshot and a boolean indicating whether the
+// snapshot is bootstrapped or not.
+func (s *ShardBlockStateSnapshot) Snapshot() (BootstrappedBlockStateSnapshot, bool) {
```

Comment:
Suggestion: Can we rename this method something like `Value()` or `State()` or similar?  I assumed looking at call sites that this was acquiring a lock or snapshotting some state, rather than just returning the immutable field value.